hax/treemakers/se_treemaker.py

```diff
@@ -0,0 +1,64 @@
+import hax
+import numpy as np
+from pax import units
+
+
+def yield_single_electrons(event, nhits_bounds, width_bounds):
+    # Get all non-lone-hit peaks in the TPC
+    peaks = [p for p in event.peaks if p.detector == 'tpc' and not p.type == 'lone_hit']
+    peaks = sorted(peaks, key=lambda p: p.left)
+
+    if not len(peaks):
+        return []
+
+    # For each of these, find the smallest gap
+    lefts, rights = np.array([(p.left, p.right) for p in peaks]).T * 10 * units.ns
+    gap_on_left = np.concatenate(([0], lefts[1:] - rights[:-1]))
+    gap_on_right = np.concatenate((lefts[1:] - rights[:-1], [0]))
+    smallest_gap = np.clip(gap_on_left, 0, gap_on_right)
+
+    # ses = []
+    # time_to_nearest = []
+    for i, peak in enumerate(peaks):
+        time_to_nearest_peak = smallest_gap[i]
+        if time_to_nearest_peak < 10 * units.us:
+            continue
+        if not (nhits_bounds[0] <= peak.n_hits < nhits_bounds[1]):
+            continue
+        width = peak.range_area_decile[5]
+        if not (width_bounds[0] <= width < width_bounds[1]):
+            continue
+        yield peak, time_to_nearest_peak
+
+
+class IsolatedPeaks(hax.minitrees.MultipleRowExtractor):
+    __version__ = '0.1.2'
+    extra_branches = ['peaks.left', 'peaks.right', 'peaks.n_hits',
+                      'peaks.n_contributing_channels',
+                      'peaks.reconstructed_positions*']
+
+    nhits_bounds = (0, float('inf'))
+    width_bounds = (0, float('inf'))
+
+    def extract_data(self, event):
+        results = []
+        for peak, time_to_nearest_peak in yield_single_electrons(event, self.nhits_bounds, self.width_bounds):
+            result = dict({x: getattr(peak, x)
+                           for x in ['area', 'area_fraction_top', 'n_hits']})
+            result['time_to_nearest_peak'] = time_to_nearest_peak
+            result['range_50p_area'] = peak.range_area_decile[5]
+            result['n_contributing_channels'] = peak.n_contributing_channels
+            for rp in peak.reconstructed_positions:
+                if rp.algorithm == 'PosRecTopPatternFit':
+                    result['x'] = rp.x
+                    result['y'] = rp.y
+                    result['xy_gof'] = rp.goodness_of_fit
+                    break
+            results.append(result)
+
+        return results
+
+
+class SingleElectrons(IsolatedPeaks):
```

Comment:
Please add a docstring explaining what this class does.  Title and few sentences describing it.